File: week05/snippets/opa_properties.mjs
import fs from 'node:fs';
import BigJSON from 'big-json';

// Load the data from the GeoJSON file
const data = await BigJSON.parse({
  body: fs.readFileSync('../exercises/opa_properties_public.geojson')
});

// Write the data to a JSONL file
const f = fs.createWriteStream('../exercises/opa_properties_public.jsonl');
for (const feature of data.features) {
  const row = feature.properties;
  row.geog = JSON.stringify(feature.geometry);
  f.write(JSON.stringify(row) + '\n'); // '\n' is to make a new line for each write
}

File: week05/snippets/opa_properties.py
import json


# Load the data from the GeoJSON file
with open('opa_properties_public.geojson', 'r', encoding='utf-8') as f:
    data = json.load(f)


# Write the data to a JSONL file
# Normally, without empty geog
# with open('opa_properties_public.jsonl', 'w') as f:
#     for feature in data['features']:
#         row = feature['properties']
#         row['geog'] = json.dumps(feature['geometry'])
#         f.write(json.dumps(row) + '\n')

with open('opa_properties.jsonl', 'w') as f:
    for feature in data['features']:
        row = feature['properties']
        if feature['geometry'] and feature['geometry']['coordinates']:
            row['geog'] = json.dumps(feature['geometry'])
        else:
            row['geog'] = None
        f.write(json.dumps(row) + '\n')
